### PR TITLE
Fixed static member access

### LINKED ISSUE #1081: [C++ verification] nonexistant member to instantiate data member with dependent non-type param
./esbmc-new cbmc/Templates23/main.cpp
```
Generating GOTO Program
ERROR: Looking up index of nonexistant member "c:@S@A>#Vi10@n" in struct/union "A<10>"
Aborted (core dumped)
```

version: ee286d92a125aa7fc3a966fc4ca85dfcd057ef9b

### PR DESCRIPTION
**Pull request by kunjsong01**
Fixed https://github.com/esbmc/esbmc/issues/1081 
generalized the conversion flow for static member access. 

Enabled 1x TC- `cbmc/Templates23`

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]

File: src/clang-c-frontend/clang_c_convert.cpp [hunk 1]
```diff
@@ -1610,26 +1610,35 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
...
         new_expr = member_exprt(base, comp.name(), comp.type());
       }
       else
       {
-        // for MemberExpr in referring to a static member
-        // which is essentially a VarDecl
+        // if a MemberExpr refers to a static member decl,
+        // we directly get the member decl symbol.
+        exprt comp;
+        if(get_decl(*member.getMemberDecl(), comp))
```
Comment (mikhailramalho) [inline src/clang-c-frontend/clang_c_convert.cpp]: this if is duplicated in both branches. It should probably be moved to outside the branches.

Comment (mikhailramalho) [inline src/clang-c-frontend/clang_c_convert.cpp]: this if is duplicated in both branches. It should probably be moved to outside the branches.

Comment (kunjsong01) [reply src/clang-c-frontend/clang_c_convert.cpp]: Done. Thanks

File: src/clang-c-frontend/clang_c_convert.cpp [hunk 2]
```diff
@@ -1608,28 +1608,31 @@ bool clang_c_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     const clang::MemberExpr &member =
       static_cast<const clang::MemberExpr &>(stmt);
 
+    exprt comp;
+    if(get_decl(*member.getMemberDecl(), comp))
```
Comment (mikhailramalho) [inline src/clang-c-frontend/clang_c_convert.cpp:1612]: I think the if I mentioned should be placed inside the `if (!perform_virtual_dispatch)`, as comp is never used in the other side of that branch.

Comment (kunjsong01) [reply src/clang-c-frontend/clang_c_convert.cpp:1612]: Somehow I missed the email notification for this comment. I'll fix it in the next patch.

**Pull request merged**